From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Wed, 14 Feb 2024 23:56:32 +0100
Subject: [PATCH 2/3] ppdCreatePPDFromIPP2(): Also check for
 "pwg-raster-document-type-supported"

If the printer supports printing input in PWG Raster format, it should
not only tell that it accepts PWG Raster in
"document-format-supported" but also provide the attributes with the
specifications of what raster formats are exactly supported, and here
not only resolutions via "pwg-raster-document-resolution-supported"
but also color spaces and depths via
"pwg-raster-document-type-supported". If one of these two attributes
is missing, we disconsider PWG Raster.

See also the discussion in

https://github.com/OpenPrinting/cups/pull/890

Thanks Zdenek Dohnal and Michael Sweet.
---
 ppd/ppd-generator.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ppd/ppd-generator.c b/ppd/ppd-generator.c
index c161f3f6..ec5be195 100644
--- a/ppd/ppd-generator.c
+++ b/ppd/ppd-generator.c
@@ -831,6 +831,7 @@ ppdCreatePPDFromIPP2(char         *buffer,          // I - Filename buffer
     is_pdf = 1;
   }
   else if (cupsArrayFind(pdl_list, "image/pwg-raster") &&
+	   ippFindAttribute(supported, "pwg-raster-document-type-supported", IPP_TAG_KEYWORD) != NULL &&
 	   (attr = ippFindAttribute(supported, "pwg-raster-document-resolution-supported", IPP_TAG_RESOLUTION)) != NULL)
   {
     current_def = NULL;
